### PR TITLE
BannerSlim: Standardize dismiss functionality

### PR DESCRIPTION
**Pull request by se7en-illa**
#### What changed?

This PR introduces a change to the BannerSlim component's dismissButton functionality. Specifically:
- The dismissButton now uses the onDismiss callback.
- The component will utilize the default provided accessibility dismiss button label.
- Updated unit tests to cover the new onDismiss behavior.

#### Why?

The primary reason for this change is to standardize the onDismiss functionality across Gestalt components. There have been variations in the way dismiss functionality is implemented, and this change ensures consistency and enhances maintainability. By centralizing the functionality to always use onDismiss, we:
- Improve the developer experience by providing a consistent API.
- Ensure accessibility standards are met by consistently applying the default dismiss button label.
-
#### Breaking Change

This change requires renaming dismissButton props in each usage. Here are codemods for this:

###### detectManualReplacement
`yarn codemod detectManualReplacement ~/path/to/your/code
--component=BannerSlim
--prop=dismissButton`

###### modifyProp
`yarn codemod modifyProp ~/path/to/your/code
--component=ActivationCard
--previousProp=dismissButton
--nextProp=onDismiss`

### Links

- [Jira](https://jira.pinadmin.com/browse/CPW-6879)
- [TDD](https://docs.google.com/document/d/13PtL_APTwvbeN_iI-Kr1irdESShA8jXWptC09ZHiDuM/edit#heading=h.qvamkklezmc7)

### CODE REVIEW

File: packages/gestalt/src/BannerSlim.tsx [hunk 1]
```diff
@@ -263,7 +241,14 @@ export default function BannerSlim({
                 </Box>
               )}
 
-              {dismissButton && <DismissButton {...dismissButton} />}
+              {onDismiss && (
+                <IconButton
+                  accessibilityLabel={accessibilityDismissButtonLabel}
+                  icon="cancel"
+                  iconColor="darkGray"
+                  onClick={onDismiss}
```
Comment (AlbertCarreras) [inline packages/gestalt/src/BannerSlim.tsx:249]: Screenshot by Dropbox Capture

File: packages/gestalt/src/BannerSlimEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+import ButtonLink from './ButtonLink';
+import { useDefaultLabelContext } from './contexts/DefaultLabelProvider';
+import Flex from './Flex';
+import Icon from './Icon';
+import IconButton from './IconButton';
+import Link from './Link';
+import MESSAGING_TYPE_ATTRIBUTES from './MESSAGING_TYPE_ATTRIBUTES';
+import Text from './Text';
+
+type DismissButtonType = {
```
Comment (AlbertCarreras) [inline packages/gestalt/src/BannerSlimEdit.tsx:13]: whats this file?

Comment (se7en-illa) [reply packages/gestalt/src/BannerSlimEdit.tsx]: test file, removed

**Comment by AlbertCarreras** [comment]
1. Before and after screenshots in PR description
2. Codemod references in PR descrption

https://github.com/pinterest/gestalt/pull/3399

File: packages/gestalt/src/BannerSlim.tsx [hunk 3]
```diff
@@ -122,6 +97,7 @@ type Props = {
    *
    */
   message: string | ReactElement;
+  onDismiss?: () => void;
```
Comment (AlbertCarreras) [inline packages/gestalt/src/BannerSlim.tsx:100]: can we keep a descriotion?

Comment (se7en-illa) [reply packages/gestalt/src/BannerSlim.tsx:103]: added description in new commit

**Comment by AlbertCarreras** [comment]
Mind updating this? 

Screenshot by Dropbox Capture

**Review by AlbertCarreras — APPROVED** [review]
Approved with final comment.

Merge after

**Pull request merged**